dev/com.ibm.ws.concurrent.persistent/src/com/ibm/ws/concurrent/persistent/internal/PersistentExecutorImpl.java

```diff
@@ -2293,12 +2295,29 @@ private long coordinateNextPoll(Config config) {
 
                 tranMgr.begin();
                 try {
-                    // TODO implement this method. For now, we invoke some basic db operations to demonstrate that what we have so far is working
                     Object[] expiryAndLastUpdated = taskStore.findPollInfoForUpdate(pollPartitionId);
                     long expiry = (Long) expiryAndLastUpdated[0];
                     long lastUpdated = (Long) expiryAndLastUpdated[1];
-                    taskStore.updatePollInfo(pollPartitionId, System.currentTimeMillis() + config.pollInterval);
-                    delay = config.pollInterval;
+                    long now = System.currentTimeMillis();
+                    int slot = 0;
+                    final int missedPollsThreshold = 2; // Could be make configurable in the future.
+
+                    if (now - lastUpdated > missedPollsThreshold * config.pollInterval) {
```

Comment:
If the current poll itself is slightly late, this could accidentally push us past the threshold and trigger the logic after just one missed poll.  We could guard against this by adding half a poll interval:
```
if (now - lastUpdated > missedPollsThreshold * config.pollInterval + config.pollInterval / 2) {
```